### PR TITLE
RHOAIENG-12535: Setup Jupyter 4 + Elyra environment on RHOAI/ODH

### PR DESCRIPTION
**Pull request by paulovmr**
## Description
* Prepare the notebooks images to use the next `odh-elyra` release.
* Remove commented code for Elyra patches that are being applied directly on ODH Elyra fork [here](https://github.com/opendatahub-io/elyra/pull/34).

* To do:
  - Confirm next version for `odh-elyra` release on PyPI
  - Update lock files

## Merge criteria:

- [X] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: jupyter/datascience/ubi9-python-3.11/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ COPY Pipfile.lock ./
 
 # TO DO: Uncomment this lines when elyra compatibility with jupyterlab 4.2 is done
```
Comment (atheo89) [inline jupyter/datascience/ubi9-python-3.11/Dockerfile:19]: Could you remove as well the comment `# TO DO: Uncomment this lines when elyra compatibility with jupyterlab 4.2 is done`

File: jupyter/datascience/ubi9-python-3.11/Dockerfile [hunk 2]
```diff
@@ -53,35 +53,16 @@ ENV PATH="$PATH:/opt/mssql-tools18/bin"
...
-  #  # Remove Elyra logo from JupyterLab because this is not a pure Elyra image \
-  #  sed -i 's/widget\.id === \x27jp-MainLogo\x27/widget\.id === \x27jp-MainLogo\x27 \&\& false/' /opt/app-root/share/jupyter/labextensions/@elyra/theme-extension/static/lib_index_js.*.js && \
-  #  # Replace Notebook's launcher, "(ipykernel)" with Python's version 3.x.y \
-RUN sed -i -e "s/Python.*/$(python --version | cut -d '.' -f-2)\",/" /opt/app-root/share/jupyter/kernels/python3/kernel.json && \
-  # TO DO: Uncomment this lines when elyra compatibility with jupyterlab 4.2 is done
-  #  # Remove default Elyra runtime-images \
-  #  rm /opt/app-root/share/jupyter/metadata/runtime-images/*.json && \
-  #  # Fix permissions to support pip in Openshift environments \
+RUN mkdir /opt/app-root/runtimes && \
+    sed -i -e "s/Python.*/$(python --version | cut -d '.' -f-2)\",/" /opt/app-root/share/jupyter/kernels/python3/kernel.json && \
```
Comment (atheo89) [inline jupyter/datascience/ubi9-python-3.11/Dockerfile:58]:
```suggestion
    # Replace Notebook's launcher, "(ipykernel)" with Python's version 3.x.y \
    sed -i -e "s/Python.*/$(python --version | cut -d '.' -f-2)\",/" /opt/app-root/share/jupyter/kernels/python3/kernel.json && \
```
Let's keep the comment for this line :slightly_smiling_face:

**Review by atheo89 — COMMENTED** [review]
/lgtm 

Just two comments to revert some comments back :slightly_smiling_face:

**Comment by jiridanek** [comment]
The CI fails are legit, please fix CI before removing the wip label.

(Just noticed we have two different wip labels in this project defined ;)

**Comment by atheo89** [comment]
Hey Paulo, could you rebase this PR and update also pipfile for TrustyAI notebook?

**Review by harshad16 — CHANGES_REQUESTED** [review]
for rocm-tensorflow image, can you please include the 
utils file https://github.com/opendatahub-io/notebooks/tree/main/jupyter/tensorflow/ubi9-python-3.11/utils and include the Dockerfile https://github.com/opendatahub-io/notebooks/blob/1600f89300c5c0ed709fbe5d807609de397a3b83/jupyter/tensorflow/ubi9-python-3.11/Dockerfile#L23C13-L23C84
so that those change can be fixed in rocm-tensorflow.

**Review by harshad16 — APPROVED** [review]
/lgtm
/approve

Reason to approve:
- The changes bring in elyra back, so it can tested in proper manner
- The PR builds get override, if new changes are pushed, so lets get this one in.
- another note: as presubmit and postsubmit jobs get different cluster, some build would be faster in the main than in PR

**Comment by paulovmr** [comment]
@jstourac Good point, since I'm rebasing it to fix conflicts, I'm also updating `odh-elyra` to use `4.0.0.dev1` version.

**Comment by atheo89** [comment]
/override ci/prow/images
Since it fails on ubi8 that we recently removed, the rest looks good
/override ci/prow/notebooks-ubi9-e2e-tests
/override ci/prow/rocm-notebooks-e2e-tests 
/lgtm

**Pull request merged**